File: ruby/bob/bob.rb
module Remark
  def silent?
    self.strip.length == 0
  end

  def shouting?
    self == self.upcase && self != self.downcase
  end

  def question?
    self.strip.end_with?('?')
  end
end

class Bob
  def self.hey remark 
    remark.extend Remark 
    case
    when remark.silent?
      "Fine. Be that way!"
    when remark.shouting?
      "Whoa, chill out!"
    when remark.question?
      "Sure."
    else
      "Whatever."
    end
  end
end

module BookKeeping
  VERSION = 1
end

File: python/bob/bob.py
def hey(phrase):
    # i miss ruby
    phrase = phrase.strip()
    if len(phrase) == 0:
        return "Fine. Be that way!"
    elif phrase[-1] == '?':
        phrase = phrase[:-1]
        if phrase.isdigit():
            return "Sure."
        if phrase == phrase.upper():
            if phrase == phrase.lower():
                return "Sure."
            else:
                return "Calm down, I know what I'm doing!"
        else:
            return "Sure."
    
    if phrase == phrase.upper():
        if phrase == phrase.lower():
            return "Whatever."
        else:
            return "Whoa, chill out!"
    else:
        return "Whatever."
